go-controller/pkg/cluster/node.go

```diff
@@ -150,47 +156,49 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 	return err
 }
 
-func updateOVNConfig(ep *kapi.Endpoints) {
-	masterIPList, southboundDBPort, northboundDBPort, err := util.ExtractDbRemotesFromEndpoint(ep)
+func updateOVNConfig(ep *kapi.Endpoints, readyChan chan bool) error {
+	masterIP, southboundDBPort, northboundDBPort, err := util.ExtractDbRemotesFromEndpoint(ep)
 	if err != nil {
-		logrus.Errorf(err.Error())
-		return
+		return err
 	}
 
-	err = config.UpdateOVNNodeAuth(masterIPList, strconv.Itoa(int(southboundDBPort)), strconv.Itoa(int(northboundDBPort)))
-	if err != nil {
-		logrus.Errorf(err.Error())
-		return
+	if err := config.UpdateOVNNodeAuth(masterIP, strconv.Itoa(int(southboundDBPort)), strconv.Itoa(int(northboundDBPort))); err != nil {
+		return err
 	}
+
 	for _, auth := range []config.OvnAuthConfig{config.OvnNorth, config.OvnSouth} {
 		if err := auth.SetDBAuth(); err != nil {
-			logrus.Errorf(err.Error())
-			return
+			return err
 		}
-		logrus.Infof("OVN databases reconfigured, masterIP %s, northbound-db %d, southbound-db %d", ep.Subsets[0].Addresses[0].IP, northboundDBPort, southboundDBPort)
+		logrus.Infof("OVN databases reconfigured, masterIP %s, northbound-db %d, southbound-db %d", masterIP, northboundDBPort, southboundDBPort)
 	}
 
+	readyChan <- true
```

Comment:
@alexanderConstantinescu you can just `close(readyChan)`

Reply:
I am not sure why this is not working while `readyChan <- true` is. I am getting a panic for closing a closed channel if the endpoint is deleted multiple times. I kind of expected that, but I am not sure why it works if I send true on the channel. The channel is buffered to one...it should not be possible to do if we don't have a go-routine constantly reading from it (which I think we don't as the only place receiving from the channel is in `node.go` L86. Where you left your previous comment.